core/controllers/admin.py

```diff
@@ -635,6 +635,166 @@ def get(self):
         self.render_json(response)
 
 
+class AddCommunityReviewerHandler(base.BaseHandler):
+    """Handles adding reviewer for community dashboard page."""
+
+    GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
+
+    @acl_decorators.can_access_admin_page
+    def post(self):
+        new_reviewer_username = self.payload.get('username')
+        new_reviewer_user_id = (
+            user_services.get_user_id_from_username(new_reviewer_username))
+
+        if new_reviewer_user_id is None:
+            raise self.InvalidInputException(
+                'Invalid username: %s' % new_reviewer_username)
+
+        review_item = self.payload.get('review_item')
+
+        if review_item == constants.REVIEWABLE_ITEM_TRANSLATION:
+            language_code = self.payload.get('language_code')
+            if utils.is_supported_audio_language_code(language_code):
+                if user_services.can_review_translation_suggestions(
+                        new_reviewer_user_id, language_code=language_code):
+                    raise self.InvalidInputException(
+                        'User %s already has rights to review translation in '
+                        'language code %s' % (
+                            new_reviewer_username, language_code))
+                else:
+                    user_services.allow_user_review_translation_in_language(
+                        new_reviewer_user_id, language_code)
+            else:
+                raise self.InvalidInputException(
+                    'Invalid language_code: %s' % language_code)
```

Comment:
Consider refactoring this to use guard clauses to reduce nesting, e.g.

```suggestion
            if not utils.is_supported_audio_language_code(language_code):
                raise self.InvalidInputException(
                    'Invalid language_code: %s' % language_code)
            if user_services.can_review_translation_suggestions(
                    new_reviewer_user_id, language_code=language_code):
                raise self.InvalidInputException(
                    'User %s already has rights to review translation in '
                    'language code %s' % (
                        new_reviewer_username, language_code))
            user_services.allow_user_review_translation_in_language(
                new_reviewer_user_id, language_code)
```

Here and other areas where the nesting is more than 2 clauses deep.

Reply:
Done!